docs/Writing-Tests.md

```diff
@@ -1,182 +1,64 @@
- - Update e2e test cases as addon functionality is changed or added 
- - Ensure the assertions and logs output by tests reflect 
-   - the execution paths and 
-   - potential causes of failures  
-   
-It is highly recommended to read over the [kubernetes best practices](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md ) for writing e2e tests. These principles should be taken into consideration for every new or existing test modified.
+Tests can be consumed by osde2e test framework in one of two ways:
 
+1. Runs tests outside of osde2e using its test harness runner (recommended)
+2. Runs tests that reside within osde2e (legacy mode/not recommended)
 
-## Writing first test
+For any new tests, they should be written in which osde2e consumes them using
+its test harness runner feature. *This applies to all OSD SREP operator tests.*
 
-### Informing vs. Blocking
-There are different suites of tests within OSDe2e with varying meanings and purposes. However, all new tests added should fall under the "informing" test suite.
+To learn more about the test harness runner of osde2e, refer to the following
+[document](Test-Harnesses.md).
 
-This test suite is used as a proving ground for new tests to validate their quality and to ensure a potentially new flaky test does not impact the overall CI Signal.
+## Test Standards "Best Practices"
 
-Once a test has run for over a week with quality results, it can then be graduated into its correct/respective suite.
+When writing new tests or enhancing existing tests, every test should strive
+to comply to the following standards:
 
-### Adding a new package of tests
-All Ginkgo tests that are imported in **[`/cmd/osde2e/test/cmd.go`]** are ran as part of the osde2e suite.
+* Follow the [kubernetes best practices guide] when writing end to end tests.
+* Review [Ginkgo] and [Gomega] as these are the core test frameworks when
+  writing OSD SREP operator tests.
+* Use [osde2e-common] module as much as possible when writing test cases. This
+  module provides common modules when working with Managed OpenShift. Which
+  aim to reduce code duplication across tests. Examples of this are:
+  clients for interfacing with OCM, OpenShift/prometheus client and much more.
+* Use the [e2e-framework] as much as possible and become familiar with it
+  when interfacing with OpenShift clusters.
+* Apply labels "tags" to your test cases allowing for easy classification
+  "grouping" of tests. This is helpful when certain tests would like to be run
+  over the entire test suite.
+* Keep test cases focused on their specific scope. Test cases are best to be
+  mapped to a given feature/functionality for the product or OSD SREP operator.
+* Ensure both positive/negative cases are covered for your test case.
+* Ensure every test case has proper error messages logged with using [Gomega]
+  matchers. This helps when it comes to troubleshooting/debugging failing tests.
```

Comment:
can we add a link to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md#debuggability on this point

Reply:
Great point, will include this link!

Reply:
Resolved by e2ad99930bdcb219570514fa02716cd1902c0262